### PR TITLE
Fix tight loop if watched etcd key does not exist.

### PR DESCRIPTION
**Pull request by fasaxc**
Required promotion of etcd stub library to avoid order dependence in UTs.

### CODE REVIEW

File: calico/openstack/test/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from calico.test import lib
```
Comment (alexwlchan) [inline calico/openstack/test/__init__.py]: This file could do with a copyright statement, to match `calico/felix/test/__init__.py`.

File: calico/test/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from calico.test import lib
```
Comment (alexwlchan) [inline calico/test/__init__.py]: Ditto above.

File: calico/openstack/test/__init__.py (hunk 1 above)
Comment (fasaxc) [inline calico/openstack/test/__init__.py]: Fixed

File: calico/test/__init__.py (hunk 2 above)
Comment (fasaxc) [inline calico/test/__init__.py]: Fixed

**Comment by plwhite** [comment]
LGTM

**Comment by plwhite** [comment]
Good for merge once the UTs have run.

**Comment by plwhite** [comment]
Sorry, this never completed tests, and now does not merge! Can you please resolve the conflicts? I'm happy to have you just merge it if the tests run clean for you.

**Comment by fasaxc** [comment]
Rebased onto master.  Interestingly, there weren't any conflicts so I'm not sure what Github was moaning about.

**Pull request merged**